playbooks/continuous_delivery/ci-cd-elements.adoc

```diff
@@ -0,0 +1,252 @@
+
+In traditional environments, the last three items are usually the responsibility of the operations team, and the developer team is not necessarily involved when they are performed. With immutable infrastructure and containers, deployment processes need to be radically rethought and need to involve the developer teams. These types of process change sometimes get push back.
+
+The phases of the progressing locking process that we have described can implemented in a CI/CD pipeline. The following sections discuss the types of pipelines and the types of steps that are part of a pipeline.
+
+== Types of pipelines
+
+Release management processes typically feature automated deployment procedures. An authorized user selects a software version and an environment, and then initiates the release process. 
+
+If the promotion workflow from the development environment to the production environment is a series of manually triggered deployments, this is referred to as a *point and click* pipeline.
+
+image::images/PointAndClick.png[]
+
+The more modern *straight through* pipeline takes code from version control system (VCS) and promotes it through the various environments to production with minimal or no human intervention.
+
+image::images/TruePipeline.png[]
+
+A straight through pipeline is generally preferred because it minimizes human error, but it requires a higher level of organization maturity.
+
+There is a continuum between strict point and click pipelines and straight through pipelines and each organization must determine the type of deployment pipeline that will fit their environment.
+
+== Ownership of the pipeline
+
+It is important to identify who in the organization owns the release process, tools, and pipelines. For traditional https://en.wikipedia.org/wiki/ITIL[ITIL-based] organizations, there is typically a release management team that owns the release process. That team might also own the tools involved in supporting the pipelines, and, in some cases,  own the actual pipelines.
+
+Ownership should be clearly defined for all pipeline-related aspects:
+
+* Release policies and compliance rules
+* Templates and other general reusable pipeline artifacts
+* Tools needed to support the pipeline
+* Individual pipelines
+
+The current trend for agile organizations is to have developer teams own the pipelines for their specific applications. In this case, the development teams are responsible for creating and updating the pipelines, as well as ensuring pipeline compliance with enterprise-issued release management standards.
+
+Because pipeline processes can be described *as code*, developers must learn the pipeline https://en.wikipedia.org/wiki/Domain-specific_language[Domain Specific Language] (DSL), but otherwise have a short learning curve.
+
+More traditional organizations are instead not comfortable with this approach because they want to have a clear separation of duty between development activities and deployment tasks.
```

Comment:
Citation needed. ;)

In all seriousness, I think this section should focus on DevOps over ITIL practices. You mention this briefly ("The current trend for agile organizations..."), and if it were up to me, I'd make that bit the focus of this section.

Reply:
I unfortunately don't have the bandwidth to rewrite this section. Sorry I can't be more helpful!

Reply:
@raffaelespazzoli @rdebeasi I think we could resolve this by adding a reference to http://v1.uncontained.io/playbooks/fundamentals/openshift_roles_responsibilities.html in this section.